storage/grpc_client.go

```diff
@@ -152,7 +152,12 @@ func (c *grpcStorageClient) GetServiceAccount(ctx context.Context, project strin
 	return resp.EmailAddress, err
 }
 
-func (c *grpcStorageClient) CreateBucket(ctx context.Context, project, bucket string, attrs *BucketAttrs, _ *bool, opts ...storageOption) (*BucketAttrs, error) {
+func (c *grpcStorageClient) CreateBucket(ctx context.Context, project, bucket string, attrs *BucketAttrs, enableObjectRetention *bool, opts ...storageOption) (*BucketAttrs, error) {
+	if enableObjectRetention != nil {
+		// TO-DO: implement ObjectRetention once available - see b/308194853
+		return nil, errors.New("storage: object retention is not supported in gRPC")
```

Comment:
Replace with codes.Unimplemented if possible. Same elsewhere that you are returning this error. https://pkg.go.dev/google.golang.org/grpc@v1.60.0/codes#Unimplemented